system-addon/test/unit/message-center/MessageCenterRouter.test.js

```diff
@@ -0,0 +1,181 @@
+    });
+  });
+  describe("#onMessage: UNBLOCK_MESSAGE_BY_ID", () => {
+    it("should remove the id from the blockList", async () => {
+      await Router.onMessage(createRemoteMessage({type: "BLOCK_MESSAGE_BY_ID", data: {id: "foo"}}));
+      assert.isTrue(Router.state.blockList.foo);
+      await Router.onMessage(createRemoteMessage({type: "UNBLOCK_MESSAGE_BY_ID", data: {id: "foo"}}));
+
+      assert.isUndefined(Router.state.blockList.foo);
+    });
+  });
+
+  describe("#onMessage: ADMIN_CONNECT_STATE", () => {
+    it("should send a message containing the whole state", async () => {
+      const msg = createRemoteMessage({type: "ADMIN_CONNECT_STATE"});
+      await Router.onMessage(msg);
+
+      assert.calledWith(msg.target.sendAsyncMessage, PARENT_TO_CHILD_MESSAGE_NAME, {type: "ADMIN_SET_STATE", data: Router.state});
+    });
+  });
+});
+
+describe("MessageCenterFeed", () => {
+  let Router;
+  let feed;
+  beforeEach(() => {
+    Router = new _MessageCenterRouter();
+    feed = new MessageCenterFeed({router: Router});
+  });
+  it("should set .router to the MessageCenterRouter singleton by default", () => {
+    feed = new MessageCenterFeed();
+
+    assert.equal(feed.router, MessageCenterRouter);
+  });
+  describe("#onAction: INIT", () => {
+    it("should initialize the MessageCenterRouter if it is not initialized and the experiment pref is true", () => {
+      // Router starts out not initialized
+      sinon.stub(Router, "init");
+
+      // Set pref to true
+      const prefs = {};
+      prefs[EXPERIMENT_PREF] = true;
+      const channel = new FakeRemotePageManager();
+      feed.store = {_messageChannel: {channel}, getState: () => ({Prefs: {values: prefs}})};
+
+      // call .onAction with INIT
+      feed.onAction({type: at.INIT});
+
+      assert.calledWith(Router.init, channel);
+    });
+    it("should not re-initialize the MessageCenterRouter if it is already initialized", () => {
+      // Router starts initialized
+      Router.init(new FakeRemotePageManager());
+      sinon.stub(Router, "init");
+
+      // Set pref to true
+      const prefs = {};
+      prefs[EXPERIMENT_PREF] = true;
+      const channel = new FakeRemotePageManager();
+      feed.store = {_messageChannel: {channel}, getState: () => ({Prefs: {values: prefs}})};
+
+      // call .onAction with INIT
+      feed.onAction({type: at.INIT});
+
+      assert.notCalled(Router.init);
+    });
+  });
+  describe("#onAction: PREF_CHANGE", () => {
+    it("should uninitialize the MessageCenterRouter if it is already initialized and the experiment pref is false", () => {
+      // Router starts initialized
+      Router.init(new FakeRemotePageManager());
+      sinon.stub(Router, "uninit");
+
+      // Set pref to true
+      const prefs = {};
+      prefs[EXPERIMENT_PREF] = false;
+      const channel = new FakeRemotePageManager();
+      feed.store = {_messageChannel: {channel}, getState: () => ({Prefs: {values: prefs}})};
```

Comment:
Could this setup go in the beforeEach? Look like the only thing that changes between the tests is the pref value of `EXPERIMENT_PREF` so maybe only expose that to each test to set its own value.